src/material/table/table.ts

```diff
@@ -32,4 +32,7 @@ import {ChangeDetectionStrategy, Component, ViewEncapsulation} from '@angular/co
 export class MatTable<T> extends CdkTable<T> {
   /** Overrides the sticky CSS class set by the `CdkTable`. */
   protected stickyCssClass = 'mat-table-sticky';
+
+  /** Overrides the need to add position: sticky on every sticky cell element in `CdkTable`. */
+  protected needsPositionStickyOnElement = false;
```

Comment:
@andrewseguin is there any reason we couldn't just _always_ use a css class for this instead of attaching styles directly?

Reply:
I'm up for making that change now, though I fear it could be breaking for any projects that have subclassed CdkTable.

Reply:
Yeah, it feels to me like an internal implementation detail, but I'm curious to get @andrewseguin's take

Reply:
I don't think there's a reason - we should have a follow up PR that attempts to switch to just the class